flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/WindowJoinUtil.scala

```diff
@@ -182,26 +183,65 @@ object WindowJoinUtil {
     }
 
     // Validate join
+    def getLeftFieldNames() = join.getLeft.getRowType.getFieldNames.toList
+
+    def getRightFieldNames() = join.getRight.getRowType.getFieldNames.toList
+
     if (windowStartEqualityLeftKeys.nonEmpty && windowEndEqualityLeftKeys.nonEmpty) {
       if (
         leftWindowProperties.getTimeAttributeType != rightWindowProperties.getTimeAttributeType) {
+
+        def timeAttributeTypeStr(isRowTime: Boolean): String = {
+          if (isRowTime) "ROWTIME" else "PROCTIME"
```

Comment:
We can just print the logical type, because we will support TIMESTAMP_LTZ as time attribute soon, so the logical type maybe different.